.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+name: Continuous Integration
+on:
+  push:
+    branches:
+#      - 'main'
+#      - 'master'
+#      - 'refs/heads/v[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+jobs:
+  supported-versions-matrix:
+    name: Supported Versions Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.supported-versions-matrix.outputs.version }}
+    steps:
+      - uses: actions/checkout@v1
+      - id: supported-versions-matrix
+        uses: WyriHaximus/github-action-composer-php-versions-in-range@v1
+  tests:
+    services:
+      postgres:
+        image: postgres:${{ matrix.postgres }}
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+    name: Testing on PHP ${{ matrix.php }} with ${{ matrix.composer }} dependency preference against Postgres ${{ matrix.postgres }}
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ${{ fromJson(needs.supported-versions-matrix.outputs.version) }}
```

Comment:
Note that this tests a much larger number of PHP versions than the old Travis setup did, is this intended? For ReactPHP we chose to explicitly list the PHP versions https://github.com/reactphp/reactphp/pull/442

Reply:
> Note that this tests a much larger number of PHP versions than the old Travis setup did, is this intended? For ReactPHP we chose to explicitly list the PHP versions [reactphp/reactphp#442](https://github.com/reactphp/reactphp/pull/442)

And I still think it's better to automate it. However, for ReactPHP we're on top of that thing. For other repositories that might be differently. This repo hasn't been updated in a while, so having newer versions automatically tested, maybe introducing Dependabot to help with that is wise here, can help. I'll leave this one up to @mbonneau and @davidwdan.